server/matchmaker/matchmaker_queue.py

```diff
@@ -19,16 +19,70 @@ def __init__(
         self.queue_name = queue_name
         self.rating_prop = 'ladder_rating'
         self.queue = OrderedDict()
+        self._matches = deque()
+        self._shutdown = False
         self._logger.debug("MatchmakerQueue initialized for %s", queue_name)
 
-    def push(self, search: Search):
+    async def iter_matches(self):
+        """ Asynchronusly yields matches as they become available """
+
+        while not self._shutdown:
+            if not self._matches:
+                await asyncio.sleep(1)
```

Comment:
while not shutdown, sleep ? 
I'm not sure of what you're trying to do but it doesn't look very clean